crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -27,19 +28,47 @@ use bevy_utils::EntityHashMap;
 /// }
 ///
 /// impl MapEntities for Spring {
-///     fn map_entities(&mut self, entity_mapper: &mut EntityMapper) {
-///         self.a = entity_mapper.get_or_reserve(self.a);
-///         self.b = entity_mapper.get_or_reserve(self.b);
+///     fn map_entities<M: EntityMapper>(&mut self, entity_mapper: &mut M) {
+///         self.a = entity_mapper.map(self.a);
+///         self.b = entity_mapper.map(self.b);
 ///     }
 /// }
 /// ```
 ///
 pub trait MapEntities {
     /// Updates all [`Entity`] references stored inside using `entity_mapper`.
-    ///
-    /// Implementors should look up any and all [`Entity`] values stored within and
-    /// update them to the mapped values via `entity_mapper`.
-    fn map_entities(&mut self, entity_mapper: &mut EntityMapper);
+    fn map_entities<M: EntityMapper>(&mut self, entity_mapper: &mut M);
+}
+
+/// Any implementor of this trait knows how to map an [`Entity`] into another [`Entity`].
+///
+/// Usually this is done by using a [`EntityHashMap<Entity, Entity>`] to map the [`Entity`] references.
+/// This can be used to map [`Entity`] references from one [`World`] to another.
+pub trait EntityMapper {
+    /// Map an entity to another entity
+    fn map(&mut self, entity: Entity) -> Entity;
+}
+
+impl EntityMapper for SceneEntityMapper<'_> {
+    /// Returns the corresponding mapped entity or reserves a new dead entity ID if it is absent.
+    fn map(&mut self, entity: Entity) -> Entity {
```

Comment:
Looks like this struct already contains `map` field, so you won't be able to call `map()` method.
I would suggest to rename the method in trait into `map_entities`. Alternatively you could rename `map` field into `entities`, but I think that `map` sounds too generic for a method from a trait.

Reply:
Yeah, I prefer `map_entities` too.